scalyr_agent/builtin_monitors/journald_monitor.py

```diff
@@ -284,11 +316,27 @@ def _initialize(self):
                 "_SOURCE_REALTIME_TIMESTAMP": "timestamp",
             }
 
+        self._logger.closeMetricLog()
+        self.log_manager = LogConfigManager(
+            self._global_config,
+            JournaldLogFormatter(),
+            self._max_log_size,
+            self._max_log_rotations,
+        )
+        self.log_config = self.log_manager.get_config(".*")
+
     def run(self):
+        self.log_manager.set_log_watcher(self._log_watcher)
         self._checkpoint = load_checkpoints(self._checkpoint_file)
         self._reset_journal()
         ScalyrMonitor.run(self)
 
+    def set_log_watcher(self, log_watcher):
+        self._log_watcher = log_watcher
+
+    def _get_log_watcher(self):
```

Comment:
Add documentation mention this returns self also... for a getter I wouldn't expect this so the documenation is nice.